### PR TITLE
FAI-11476: Update Xray source spec 

### PR DESCRIPTION
**Pull request by chalenge**
## Description

- Update Xray source spec 
- Add support for failed requests 

## Type of change
- [X] Bug fix
- [X] New feature
- [ ] Breaking change

### CODE REVIEW

File: sources/xray-source/resources/spec.json [hunk 1]
```diff
@@ -2,23 +2,50 @@
...
+      "projects": {
+        "order": 3,
+        "type": "array",
+        "title": "Project Keys or IDs",
+        "description": "List of Jira project Keys or IDs to sync.",
+        "items": {
+          "type": "string"
+        }
+      },
+      "cutoff_days": {
```
Comment (ypc-faros) [inline sources/xray-source/resources/spec.json:37]: Is `cutoff_days` used?

File: sources/xray-source/src/xray.ts [hunk 2]
```diff
@@ -32,18 +33,13 @@ export class Xray {
...
-
-    if (!config.client_id || !config.client_secret) {
-      throw new VError(
-        'Please provide Xray Cloud authentication details, ' +
-          'Client Id and a Client Secret'
-      );
-    }
-
+    const timeout = config.timeout
+      ? Math.min(config.timeout, XRAY_DEFAULT_TIMEOUT)
```
Comment (ypc-faros) [inline sources/xray-source/src/xray.ts:37]: Why do we have to do this? If this is a limitation enforced by their api, let's document it in the spec

**Review by ypc-faros — APPROVED** [review]

File: sources/xray-source/resources/spec.json (hunk 1 above)
Comment (chalenge) [reply sources/xray-source/resources/spec.json:37]: Removed, have another PR where I will re-add it

File: sources/xray-source/src/xray.ts (hunk 2 above)
Comment (chalenge) [reply sources/xray-source/src/xray.ts:37]: Not supposed to be here 🤦🏽

**Pull request merged**